vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1656,12 +1667,22 @@ def _build_attn_group_metadata(
                 attn_metadata_i = builder.build_for_cudagraph_capture(
                     common_attn_metadata
                 )
+            elif (
+                cache_key in cached_attn_metadata
+                and builder.supports_update_block_table
+            ):
+                attn_metadata_i = builder.update_block_table(
+                    cached_attn_metadata[cache_key],
+                    common_attn_metadata.block_table_tensor,
+                    common_attn_metadata.slot_mapping,
+                )
             else:
                 attn_metadata_i = builder.build(
                     common_prefix_len=cascade_attn_prefix_len,
                     common_attn_metadata=common_attn_metadata,
                     **extra_attn_metadata_args,
                 )
+                cached_attn_metadata[cache_key] = attn_metadata_i
```

Comment:
nit: should we do it all the time or also condition it on `builder.supports_update_block_table` ?